### PR TITLE
Patch -useexedir

### LINKED ISSUE #322: [Runtime Issue]: `-useexedir` causes game to crash instantly
### Build Version

v1.5.0 7d38dde2

### Operating System Environment

- [ ] Microsoft Windows (32-bit)
- [ ] Microsoft Windows (64-bit)
- [ ] Mac OS X
- [X] Linux (specify distribution and version below)

### CPU Environment

- [ ] x86 (32-bit Intel/AMD)
- [X] x86\_64 (64-bit Intel/AMD)
- [ ] ARM (32-bit)
- [ ] ARM64 (64-bit; sometimes called AArch64)
- [ ] Other (RISC V, PPC...)

### Game Modes Affected

- [X] Single player
- [X] Anarchy
- [X] Hyper-Anarchy
- [X] Robo-Anarchy
- [X] Team Anarchy
- [X] Capture the Flag
- [X] Bounty
- [X] Entropy
- [X] Hoard
- [X] Monsterball
- [X] Cooperative

### Game Environment

I’m running NixOS 23.11 and using game data from the Windows Dual-Jewel version of Descent 3 v1.4.

### Description

Whenever I try to run the game with `-useexedir`, it crashes with these errors:

```
Error: Unable to set temporary directory to: "./Descent3/custom/cache"

Error: Unable to set temporary directory to: "./Descent3/custom/cache"(Descent 3 Error)

System Error

Press OK to exit, CANCEL to ignore this error and continue.
SIGNAL 5 caught, aborting
DDIO: ddio_InternalClose() called.
DDIO: ddio_InternalClose() returning.
```

### Regression Status

I don’t know.

### Steps to Reproduce

1. Run `Descent3 -useexedir`.

### PR DESCRIPTION
**Pull request by JeodC**
## Pull Request Type

- [ ] GitHub Workflow changes
- [ ] Documentation or Wiki changes
- [ ] Build and Dependency changes
- [x] Runtime changes
  - [ ] Render changes
  - [ ] Audio changes
  - [ ] Input changes
  - [ ] Network changes
  - [ ] Other changes

### Description

Uses `Base_directory[i]` in the loop to correctly truncate the path.

### Related Issues

#322

### CODE REVIEW

File: Descent3/init.cpp [hunk 1]
```diff
@@ -1571,7 +1571,7 @@ void InitIOSystems(bool editor) {
     strcpy(Base_directory, GameArgs[0]);
     int len = strlen(Base_directory);
     for (int i = (len - 1); i >= 0; i--) {
-      if (i == '\\') {
+      if (Base_directory[i] == '\\' || Base_directory[i] == '/') {
         Base_directory[i] = '\0';
```
Comment (Lgt2x) [inline Descent3/init.cpp:1575]: you need to exit the loop using `break;` here when the first path separator is found.

**Review by Lgt2x — APPROVED** [review]

**Comment by JeodC** [comment]
Adds a partial improvement that does not fully resolve #322. Will test with windows later and merge if successful. #322 will need further improvements.

**Comment by JeodC** [comment]
@Lgt2x This does not work in windows either:

image

But the changes are still a step in the right direction. Should this be merged anyway or left open to be worked?

**Comment by Lgt2x** [comment]
> @Lgt2x This does not work in windows either:
> 
> image
> 
> But the changes are still a step in the right direction. Should this be merged anyway or left open to be worked?

`Base_directory` seems to be wrong. Can you step with a debugger in the function to identify the problem?

**Pull request closed without merging**